### PR TITLE
ci: Attempt to fix git error

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Try and add the recommended safe diretory fix to resolve the
    `unsafe repository <repo> is owned by someone else` error

Fixes: #4698
Signed-off-by: stevenhorsman <steven@uk.ibm.com>

### CODE REVIEW

**Comment by stevenhorsman** [comment]
So removing the sudo seemed to fix the git error, but lead to another one (which is probably why the sudo was added in the first place):
```
go: writing stat cache: open /tmp/jenkins/workspace/kata-containers-CCv0-tests-ubuntu-20.04-PR/go/pkg/mod/cache/download/github.com/onsi/ginkgo/@v/v1.16.5.info298498081.tmp: permission denied
11:38:23 go: downloading github.com/onsi/ginkgo v1.16.5
11:38:23 go install: github.com/onsi/ginkgo/ginkgo@v1.16.5: open /tmp/jenkins/workspace/kata-containers-CCv0-tests-ubuntu-20.04-PR/go/pkg/mod/cache/download/github.com/onsi/ginkgo/@v/v1.16.5.lock: permission denied
```

**Comment by stevenhorsman** [comment]
I had a little look to see if I could work out how to resolve the gingko error, but gave up and thought I'd try the recommended fix of adding the directory to git's safe.directory config, so time to /test that

**Comment by wainersm** [comment]
Hi @stevenhorsman , thanks for this fix!

Question: did you try without `--global`? Asking because it would be less invasive, not writing on a global config file generally makes us developers happy :)

**Comment by stevenhorsman** [comment]
Yep, with `--local` it failed: e.g. http://jenkins.katacontainers.io/job/tests-CCv0-ubuntu-20.04-x86_64-CC_SKOPEO_CRI_CONTAINERD-PR/34/console:
```
http://jenkins.katacontainers.io/job/tests-CCv0-ubuntu-20.04-x86_64-CC_SKOPEO_CRI_CONTAINERD-PR/34/console
```
I definitely understand that `--global` is not ideal, and would rather fix it by getting to the bottom of why `sudo` is needed in our fork of containerd, but not the original, but I don't want to hold up all CC development, so I think it's a necessary evil at the moment

**Comment by wainersm** [comment]
> Yep, with `--local` it failed: e.g. http://jenkins.katacontainers.io/job/tests-CCv0-ubuntu-20.04-x86_64-CC_SKOPEO_CRI_CONTAINERD-PR/34/console:
> 
> ```
> http://jenkins.katacontainers.io/job/tests-CCv0-ubuntu-20.04-x86_64-CC_SKOPEO_CRI_CONTAINERD-PR/34/console
> ```
>

hmm... my thesis is that even if with `--local` an attacker can explore the security flaw, all in all he/she just need to have write access to local repo (whereas `--global` requires root or same user privileges). Anyway, I am not a security expert :)
  
> I definitely understand that `--global` is not ideal, and would rather fix it by getting to the bottom of why `sudo` is needed in our fork of containerd, but not the original, but I don't want to hold up all CC development, so I think it's a necessary evil at the moment

I agree with you. That's not going to be the first time scripts write on system-wide...

**Review by wainersm — APPROVED** [review]

**Pull request merged**
